Tasks/Kubernetes/src/clusterconnection.ts

```diff
@@ -0,0 +1,81 @@
+"use strict";
+
+import * as fs from "fs";
+import * as path from "path";
+import * as url from "url";
+import * as tl from "vsts-task-lib/task";
+import * as tr from "vsts-task-lib/toolrunner";
+import AuthenticationToken from "docker-common/registryauthenticationprovider/registryauthenticationtoken"
+import * as utils from "./utilities";
+import * as os from "os";
+
+export default class ClusterConnection {
+    private kubectlPath: string;
+    private kubeconfigFile: string;
+    private userDir: string;
+
+    constructor() {
+        this.kubectlPath = tl.which("kubectl", false);
+        this.userDir = utils.getNewUserDirPath();
+    }
+
+    private async initialize() {
+        if(!this.kubectlPath || !fs.existsSync(this.kubectlPath))
+        {
+            tl.debug(tl.loc("DownloadingClient"));
+            this.kubectlPath = path.join(this.userDir, "kubectl") + this.getExecutableExtention();
+            var version = await utils.getStableKubectlVersion();
```

Comment:
we can use an agrument --generator to fix the version of the rest call make by kubectl client.

https://kubernetes.io/docs/user-guide/kubectl-conventions/